components/filter/filter.js

```diff
@@ -487,15 +494,30 @@ class Filter extends FocusMixin(LocalizeCoreElement(RtlMixin(LitElement))) {
 			if (count === 0) return searchResults;
 		}
 
+		let listHeader = nothing;
+		let offscreenHeader = nothing;
+		if (dimension.headerText && dimension.searchValue === '') {
+			listHeader = html`
+				<d2l-list-item>
+					<h4 class="d2l-heading-4 list-header-text" aria-hidden="true">${dimension.headerText}</h4>
```

Comment:
After discussion, to make the header text more readable, I added an offscreen header that is rendered outside of the list. I added an `aria-hidden` here when `header-text` is set and `selected-first` is not since the header will be at the top and does not need to be listed twice in the a11y DOM. I think we do still want to render the header in the list in this case so the appearance is consistent.

When selected-first is added, the plan is to make the offscreen header read something like `${headerText}. Selected items at top.` and remove the `aria-hidden` on the list header in that case since we want the header in the a11y DOM to break up the list into 2 parts. The list header may not be read depending on how you are navigating, so the offscreen header will work as a backup which is read outside of the list.

Reply:
Gave it a try with NVDA and JAWS - they won't hit the offscreen `h4` at all if navigating with tab. If they navigate with arrow keys only, they'll hear it in single-dimension mode - it's a bit awkward though because then the list won't work properly unless they know to turn on Forms/Focus mode. In multi-dimension mode they won't hear it, I _think_ because of [DE52188](https://rally1.rallydev.com/#/15545167705ud/search?detail=%2Fdefect%2F686309407849).

If they try to find it based on the headings shortcut that works, but it's maybe a tad confusing because the other headings on the page aren't hidden when the filter is opened. Did Carin also play around with this stuff? Maybe she's navigating in a way I don't know about

Reply:
Right, if you navigate with tabbing then the issue will be the same. Even in our ideal case where we have 2 lists with labels, I don't think those labels would get read if you are tabbing. That being said, I'm not sure if screen reader users normally navigate with tabs, it seems like that would end up skipping information in most situations. 

The main idea with adding the header above the list was so that users would encounter the header text before entering the list, where they may begin navigating with arrow keys which would skip over the header. Carin hasn't had a look at this, it may be good to get her opinion and see how she would navigate.

Reply:
> Even in our ideal case where we have 2 lists with labels, I don't think those labels would get read if you are tabbing

Hmm, I wonder if that's a VoiceOver thing? If I add a `label` attribute to `d2l-list` and pass that to `aria-label` on the [`div` that sets the `role`](https://github.com/BrightspaceUI/core/blob/main/components/list/list.js#L123), and pass the `header-text` in there instead, then NVDA and JAWS will read that header text instead of "application" each time I tab to the list.

Reply:
>  That being said, I'm not sure if screen reader users normally navigate with tabs, it seems like that would end up skipping information in most situations.
> Carin hasn't had a look at this, it may be good to get her opinion and see how she would navigate.

Yeah, this might be one of those cases where how I'm testing doesn't match how a real user would behave

Reply:
This is now using the header text as a list label which seems to work nicely with a screenreader. The header itself is just used for presentation and is given `aria-hidden="true"`. 

I don't think there is a need for this to be navigable by header in the screenreader so this seems okay.

For selected first we will say: `${headerText}. Selected filters appear first.`